src/with-component.js

```diff
@@ -2,9 +2,10 @@ import { render } from 'preact';
 import { HTMLElement } from './util';
 import { withProps } from './with-props';
 import { withRender } from './with-render';
+import { withUnique } from './with-unique';
 
 export function withComponent (Base = HTMLElement) {
-  return class extends withRender(withProps(Base)) {
+  return class extends withUnique(withRender(withProps(Base))) {
```

Comment:
Yeah, I'm concerned about that, too. If we just used functions, then they could be used as decorators and not add to the chain. Normal `instanceof` checks don't work with this style of mixins without a [bit of intervention](http://justinfagnani.com/2016/01/07/enhancing-mixins-with-decorator-functions/), anyways.